core/runtime/common/runtime_upgrade_tracker_impl.hpp

```diff
@@ -38,7 +38,7 @@ namespace kagome::runtime {
    private:
     // assumption: insertions in the middle should be extremely rare, if any
     // assumption: runtime upgrades are rare
-    std::vector<primitives::BlockInfo> blocks_with_runtime_upgrade_;
+    mutable std::vector<primitives::BlockInfo> blocks_with_runtime_upgrade_;
```

Comment:
I recommend not to use mutables. Only as temporary fix.

Reply:
It's basically just a cache, and I find usage of mutables for local caches harmless (and this is perhaps the only case I can remember when it's justified).